### PR TITLE
Add runkit

### PR DESCRIPTION
**Pull request by alexmace**
Complete with spec.

Does use a namesco fork of zenovich/runkit because a tag is needed to get the source

### CODE REVIEW

File: manifests/extension/runkit.pp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#     }
+#
+define php::extension::runkit(
+  $php,
+  $version = '1.0.3'
+) {
+  include boxen::config
+
+  require php::config
+  # Require php version eg. php::5_4_10
```
Comment (sambauers) [inline manifests/extension/runkit.pp]: New syntax here please.

File: manifests/extension/runkit.pp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  php_require($php)
+
+  $extension = 'runkit'
+
+  # Final module install path
+  $module_path = "${php::config::root}/versions/${php}/modules/${extension}.so"
+
+  # Clone the source respository
+  repository { "${php::config::extensioncachedir}/runkit":
+    source => 'namesco/runkit'
```
Comment (sambauers) [inline manifests/extension/runkit.pp]: I'm not a fan of using non-canonical source. Can you get the upstream tagged by the original author?

Comment (alexmace) [inline manifests/extension/runkit.pp]: The "canonical" source for runkit is... complicated. 

The PECL extension hasn't had a release for 8 years: https://pecl.php.net/package/runkit
Even compiling from source for PECL isn't compatible with PHP 5.3+: http://git.php.net/?p=pecl/php/runkit.git;a=summary
There is a fork on GitHub which works up to PHP 5.5: https://github.com/zenovich/runkit
There is then another fork of that works on PHP 5.6: https://github.com/padraic/runkit

Basically it's a mess, I can request any of these upstreams to create a tag, but none are the canonical source.

Comment (sambauers) [inline manifests/extension/runkit.pp]: I think even getting those to make a tag would be better than forking and tagging your own. I don't know. It makes me wonder if this is really appropriate for inclusion in the first place if it is that poorly supported.

**Comment by alexmace** [comment]
That's up to you. It works as it stands, even if the extension itself, it's maintenance and it's use cases are all questionable.
